### DISCUSSION TITLE
Meeting minutes for 25.01.2023

### DISCUSSION THREAD

**Discussion by konradweiss** [Fraunhofer-AISEC/cpg#1419] [Dev Meetings]
### Agenda

1. Open PRs
2. DFG & Accesses to struct members

### Discussed PRs

- [ ] #1418 
- [ ] #1407 
- [ ] #1402 
- [ ] #1382 
- [ ] #1171 
...

**Comment by oxisto** [comment]
# Open PRs

1. #1418 Enabling execution of additional non-default passes. Looks good. @peckto will add CDG
2. #1407 We enable inference per default and place variables and methods in the AST. I will extend the AST placement in a future PR. We changed the behaviour of DFG edges in inferred functions to be more compatible with non-inferred functions and would enable the integration of function summaries
3. #1402 Feature complete, but waiting for SootUp's next maven central release
4. #1382 Paper only, not mergeable (yet)
5. #1171 We will leave out the GitHub actions part and parsing of comments, but apart from that we can merge it in

# DFG & Access to struct members

Proposal:

1. We remove the DFG edges from/to the field declarations (the "usages" field should already be populated, to find references/memberexpressions from/to this field)
2. We add a DFG edge from the member expression to its base (reference), and add a property to the DFG edge, which denotes which field is written/read to/from. This property includes a pointer to the node of the field declaration.
3. Step 2 will probably be done in the context sensitive DFG pass.

Nested structures could be a problematic.

# State of cpg-console

Still broken, because the kotlin console fixed it, but not released yet
Workaround: https://github.com/Fraunhofer-AISEC/cpg/issues/1272#issuecomment-1909454286
